spec/compiler/semantic/annotation_spec.cr

```diff
@@ -12,24 +12,375 @@ describe "Semantic: annotation" do
     type.name.should eq("Foo")
   end
 
-  it "can't find annotation in module" do
-    assert_type(%(
+  describe "#annotations" do
+    it "returns an empty array if there are none defined" do
+      assert_type(%(
+        annotation Foo
+        end
+
+        module Moo
+        end
+
+        {% if Moo.annotations(Foo).size == 0 %}
+          1
+        {% else %}
+          'a'
+        {% end %}
+      )) { int32 }
+    end
+
+    it "finds annotations on a module" do
+      assert_type(%(
+        annotation Foo
+        end
+
+        @[Foo]
+        @[Foo]
+        module Moo
+        end
+
+        {% if Moo.annotations(Foo).size == 2 %}
+          1
+        {% else %}
+          'a'
+        {% end %}
+      )) { int32 }
+    end
+
+    it "uses annotations value, positional" do
+      assert_type(%(
+        annotation Foo
+        end
+
+        @[Foo(1)]
+        @[Foo(2)]
+        module Moo
+        end
+
+        {% if Moo.annotations(Foo)[0][0] == 1 && Moo.annotations(Foo)[1][0] == 2 %}
+          1
+        {% else %}
+          'a'
+        {% end %}
+      )) { int32 }
+    end
+
+    it "uses annotations value, keyword" do
+      assert_type(%(
```

Comment:
Many code snippets are missing indentation. The code below this line, for example, but many others too.

Reply:
It looks fine to me